frappe/model/db_query.py

```diff
@@ -550,6 +556,75 @@ def build_filter_conditions(self, filters, conditions, ignore_permissions=None):
 			else:
 				conditions.append(self.prepare_filter_condition(f))
 
+	def apply_fieldlevel_read_permissions(self):
+		"""Apply fieldlevel read permissions to the query"""
+		if self.flags.ignore_permissions:
+			return
+
+		permitted_fields = get_permitted_fields(doctype=self.doctype)
+
+		for i, field in enumerate(self.fields):
+			if "distinct" in field.lower():
+				# field: 'count(distinct `tabPhoto`.name) as total_count'
+				# column: 'tabPhoto.name'
+				self.distinct = True
+				column = field.split(" ", 2)[1].replace("`", "").replace(")", "")
+			else:
+				# field: 'count(`tabPhoto`.name) as total_count'
+				# column: 'tabPhoto.name'
+				column = field.split("(")[-1].split(")", 1)[0]
+				column = strip_alias(column).replace("`", "")
+
+			if column == "*" and not in_function("*", field):
+				self.fields[i : i + 1] = permitted_fields
```

Comment:
Could cause the "mutating list while updating error" 🤔

Reply:
Don't think this affects objects other than dict & set. But if you come across this error, share context or suggest safer alternatives that we could use.

Reply:
let's assume it doesn't. you're still looping through those already permitted fields unnecessarily (`enumerate` returns an iterator).

alternatives:
- store index, do this at end of loop?
- do this at beginning of loop? you just need to check if the stripped value is "*" anyway - it can't have backticks or can't be in a function.

Reply:
Implemented first suggestion via https://github.com/frappe/frappe/pull/19533/commits/2ae6b7f016506acea8b84de168ca13823e021d72